frontend/src/lib/utils/export-to-csv.utils.ts

```
@@ -0,0 +1,127 @@
+ * - Adds BOM character for proper UTF-8 encoding in Excel
+ */
+export const downloadCSV = async ({
+  entity,
+  fileName = "entity",
+  description = " CSV file",
+}: {
+  entity: Record<string, unknown>[];
+  fileName?: string;
+  description?: string;
+}): Promise<void> => {
+  try {
+    if (!entity?.length) {
+      throw new CSVGenerationError("No data provided for CSV generation");
+    }
+
+    const csvContent = convertToCSV(entity);
+    const blob = new Blob(["\uFEFF" + csvContent], {
+      type: "text/csv;charset=utf-8;",
+    });
+
+    if (window.showSaveFilePicker) {
+      try {
+        const handle = await window.showSaveFilePicker({
+          suggestedName: `${fileName}.csv`,
+          types: [
+            {
+              description,
+              accept: { "text/csv": [".csv"] },
+            },
+          ],
+        });
+        const writable = await handle.createWritable();
+        await writable.write(blob);
+        await writable.close();
+      } catch (error) {
+        // User cancelled the save dialog - not an error
+        if (error instanceof Error && error.name === "AbortError") return;
+
+        throw new FileSystemAccessError(
+          "Failed to save file using File System Access API"
+        );
+      }
+    } else {
+      try {
+        const url = URL.createObjectURL(blob);
+        const link = document.createElement("a");
+        link.href = url;
+        link.download = "neurons.csv";
+        document.body.appendChild(link);
+        link.click();
+        document.body.removeChild(link);
+        URL.revokeObjectURL(url);
+      } catch (error) {
+        throw new FileSystemAccessError(
+          "Failed to save file using fallback method"
+        );
+      }
+    }
+  } catch (error) {
+    if (
+      error instanceof FileSystemAccessError ||
+      error instanceof CSVGenerationError
+    ) {
+      throw error;
+    }
+    throw new Error("Unexpected error during CSV download");
```

Why do we want to hide the real error if it's not an instance of `FileSystemAccessError` or `CSVGenerationError`?